Remove Quirk shouldDisablePopoverAttributeQuirk for apple-console.lrn.com

karlcow opened the pull request:
#### 45bc0fda2eb73defa784ca04826c9b7d66a9a558
<pre>
Remove Quirk shouldDisablePopoverAttributeQuirk for apple-console.lrn.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=271216">https://bugs.webkit.org/show_bug.cgi?id=271216</a>
<a href="https://rdar.apple.com/124989024">rdar://124989024</a>

Reviewed by Tim Nguyen and Anne van Kesteren.

The bug was fixed in the initial library which was used by the apple site.
On May 2023, the site owners confirmed that they had fixed the issue on
their side and adjusted the library. The Quirk can be removed.
The Quirk was added by
<a href="https://bugs.webkit.org/show_bug.cgi?id=255373">https://bugs.webkit.org/show_bug.cgi?id=255373</a>
<a href="https://rdar.apple.com/106779034">rdar://106779034</a>

* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::attributeChanged):
* Source/WebCore/html/HTMLElement.idl:
* Source/WebCore/html/HTMLFormControlElement.cpp:
(WebCore::HTMLFormControlElement::popoverTargetElement const):
* Source/WebCore/html/PopoverInvokerElement.idl:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisablePopoverAttributeQuirk const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):

Canonical link: <a href="https://commits.webkit.org/276498@main">https://commits.webkit.org/276498@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8d5e556ae30f28bbd40a7cdd1cbc5b9f56dc38fa

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44693 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23782 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47157 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47346 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40698 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46995 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27820 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21170 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36747 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45270 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20850 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38472 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17800 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18286 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39619 "Found 1 new test failure: imported/w3c/web-platform-tests/xhr/setrequestheader-case-insensitive.htm (failure)") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2740 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40919 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39905 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49014 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19653 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16210 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43717 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20978 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42460 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9984 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21311 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20648 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/html/HTMLElement.cpp:
```diff
@@ -423,7 +423,7 @@ void HTMLElement::attributeChanged(const QualifiedName& name, const AtomString&
         return;
     case AttributeNames::popoverAttr:
-        if (document().settings().popoverAttributeEnabled() && !document().quirks().shouldDisablePopoverAttributeQuirk())
+        if (document().settings().popoverAttributeEnabled() && name == popoverAttr)
```
I'm not sure why `name == popoverAttr` was added here. That's already taken care of by the `case` statement directly above.

karlcow replied:
Oh! I know where it is coming from. Starting the revert to the previous code but forgot to drop the popoverAttr. It was previously just a if. 
Thanks.

karlcow committed:
Remove Quirk shouldDisablePopoverAttributeQuirk for apple-console.lrn.com
https://bugs.webkit.org/show_bug.cgi?id=271216
rdar://124989024

Reviewed by Tim Nguyen and Anne van Kesteren.

The bug was fixed in the initial library which was used by the apple site.
On May 2023, the site owners confirmed that they had fixed the issue on
their side and adjusted the library. The Quirk can be removed.
The Quirk was added by
https://bugs.webkit.org/show_bug.cgi?id=255373
rdar://106779034

* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::attributeChanged):
* Source/WebCore/html/HTMLElement.idl:
* Source/WebCore/html/HTMLFormControlElement.cpp:
(WebCore::HTMLFormControlElement::popoverTargetElement const):
* Source/WebCore/html/PopoverInvokerElement.idl:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisablePopoverAttributeQuirk const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):

Canonical link: https://commits.webkit.org/276498@main
